### ISSUE TITLE
Can't renew certs after the DNS AAAA record has been added

### ISSUE THREAD

**Issue by lonequantum** [Mailu/Mailu#1429]
Hello,
I'm running Mailu 1.7 with "letsencrypt" as TLS_FLAVOR, since 2 months.
Last updated this morning. No tweaks nor additional settings.
Everything is OK except certificate renewal.
```
2020-04-02 12:51:56,771:DEBUG:acme.client:Storing nonce: 0001kwmdP70iCZsD4V_4oZsb0vG9I0MTiyWTiLMnit0NvdU
2020-04-02 12:51:56,775:WARNING:certbot.auth_handler:Challenge failed for domain mail.lnode.net
2020-04-02 12:51:56,777:INFO:certbot.auth_handler:http-01 challenge for mail.lnode.net
2020-04-02 12:51:56,779:DEBUG:certbot.reporter:Reporting to user: The following errors were reported by the server:

Domain: mail.lnode.net
Type:   connection
Detail: Fetching https://mail.lnode.net/.well-known/acme-challenge/q05OAyuJUx7X0lZMYgJSPdbYKqnNmJgOpmqEbsf8MLw: Connection refused
```
It seems it is related to the nginx conf:
```
      location ^~ /.well-known/acme-challenge/ {
          proxy_pass http://127.0.0.1:8008;
      }
```
AFAIK, "Connection refused" means the port is not open.

Any idea to fix/workaround this? My cert expires in 10 days… (and it's been 20 days the Python script retries every hour)
Thx!

**Comment by ofthesun9** [comment]
Hello,
Are the DNS (A and AAAA) for mail.lnode.net properly registered ? Are ports 80 & 443 open on this machine and accessible from"outside" (NAT, iptable, etc...)
You can also check the nginx config used by the fronted :
`docker-compose exec front cat /etc/nginx/nginx.conf`
 and check if the `location ^~ /.well-known/acme-challenge/` declaration is there...

**Comment by lonequantum** [comment]
Yep, all these things are OK.
Now I just have deleted the AAAA record from my zone, because I had added it AFTER Mailu successfully obtained the cert, 2 month ago. So I want to check if the AAAA record is a problem rather than a solution. Let's see in some days (time for removal propagation)…

**Comment by lonequantum** [comment]
Yeeeah it's renewed!!! :-D
Finally, no need to wait several days. And it has nothing to do with this mysterious port 8008.

The moral of this story is : if you get a certificate from LE with only IPv4 (A) in your DNS, don't add the IPv6 record (AAAA) - or at least delete it when it's time to renew.

Is it a behavior that is specific to the way Mailu manages Certbot? Or would it be a general recommendation to any Certbot installation? I don't close the issue now because of this question.

**Comment by lonequantum** [comment]
I did the same with another service I manage with [docker-haproxy-certbot](https://github.com/lonequantum/docker-haproxy-certbot).
That is, I had added the AAAA DNS record after the service was online.
But I've just tested the renewal of the cert of that service and it works well.

In short, the problem I encountered with my Mailu instance's cert renewing seems not to be related with Certbot itself.
I rename the issue title.
Mailu developers, good luck ;-)

**Comment by te-online** [comment]
Just experienced the same symptoms, but different issue, maybe it's the same for you:
I simply had an error in my `AAAA` DNS record. It wasn't the correct address.

Other possible angles:
- Maybe your docker containers are not available from your ipv6 address?
- Did you check your `docker-compose.yml`? Maybe you're binding directly to your ipv4 address?
- What happens if you run `ifconfig` on the host? Do you see the ipv6 address that is configured in your DNS mapped to the correct interface?

This might not be a problem with Mailu, but rather with your specific setup, wouldn't we otherwise see this issue a lot more often?

**Comment by lonequantum** [comment]
Yes you're right!
Indeed, I was using the web-generated `docker-compose.yml` and I had reviewed it too quickly - there was bindings to ipv4.
I added again the AAAA record in my zone (and waited some time, you never know), removed the ipv4 bindings from the file, restarted Mailu and tested the cert renewal… everything worked like a charm.

However, this leads to another question: why the hell LE doesn't try ipv4 if ipv6 fails? The situation was clear: my Mailu instance was configured for ipv4 only, but the *presence* of the AAAA, along side the A record, was blocking the process of renewing…
Anyway, it seems that finally it is more related to Certbot/LE than Mailu, so I close the issue.
Thanks!
